Documentation/patching-guidelines.md

```diff
@@ -0,0 +1,106 @@
+# Patching Guidelines
+
+There are times when the source build product will have build errors, functional
+defects, poison leaks, prebuilts, etc. Ideally, the underlying issues
+would be addressed at the source of the problem (e.g. within the product repos)
+and flow into source build. The reality is that repo fixes can take a long
+ time to flow into source build. This can block source build development
+(e.g. builds, tests, feature development, etc.). To avoid these delays source
+patches are applied within source-build to the repo source. These patches are intended
+to be ephemeral until the repo fixes flow in.
+
+This document provides guidance for creating and managing source build patches.
+
+## Creating Patches
+
+To create a repo patch file, first commit your changes to the repo as normal,
+then run this command inside the repo to generate a patch file inside the repo:
+
+```sh
+git format-patch --zero-commit --no-signature -1
+```
+
+Then, move the patch file into the [installer repo](https://github.com/dotnet/installer/tree/main/src/SourceBuild/patches)
+, at `src/SourceBuild/patches/<repo>`.
+
+> If you define `PATCH_DIR` to point at the `patches` directory, you can use
+> `-o` to place the patch file directly in the right directory:
+>
+> ```sh
+> git format-patch --zero-commit --no-signature -1 -o "$PATCH_DIR/<repo>"
+> ```
+
+To apply a patch, or multiple patches, use `git am` while inside the target
+repo. For example, to apply *all* `sdk` patches onto a fresh clone of the `sdk`
+repository that has already been checked out to the correct commit, use:
+
+```sh
+git am "$PATCH_DIR/sdk/*"
+```
+
+This creates a Git commit with the patch contents, so you can easily amend a
+patch or create a new commit on top that you can be sure will apply cleanly.
```

Comment:
This should be in a separate section labeled as "Update an Existing Patch" and should have another paragraph at the beginning describing the scenario.

Reply:
I separated this into an `Applying Patches` section.  It is useful for ensuring patches apply, amending patches, and defining a new patch that touches code modified by existing patches.  The `Resolving Patch Conflicts` section is of course related.